plugins/cue_points/quiz/lib/api/KalturaAnswerCuePoint.php

```diff
@@ -88,31 +80,63 @@ public function doFromObject($dbObject, KalturaDetachedResponseProfile $response
 		parent::doFromObject($dbObject, $responseProfile);
 
 		$dbEntry = entryPeer::retrieveByPK($dbObject->getEntryId());
-		if ( !kEntitlementUtils::isEntitledForEditEntry($dbEntry) ) {
+		if ( !kEntitlementUtils::isEntitledForEditEntry($dbEntry) )
+		{
 			/**
 			 * @var kQuiz $kQuiz
 			 */
 			$kQuiz = QuizPlugin::validateAndGetQuiz( $dbEntry );
 
+			//Find the corresponding question cue point
+			$this->validateParentId();
+			$dbQuestionCuePoint = CuePointPeer::retrieveByPK($this->parentId);
+
+			$kQuestionEntry= $this->getEntry($dbQuestionCuePoint->getEntryId(), $dbQuestionCuePoint->getType());
+			$this->explanation = $kQuestionEntry->explanation;
```

Comment:
All the lines that are relevant in getting the parent question should be encapsulated into one private function that returns the object of question after validation.